File: ClimbEdge.Application/DTOs/PaymentDTO.cs
using ClimbEdge.Domain.Enums.Payment;

namespace ClimbEdge.Application.DTOs
{
    public record class CreateSubscriptionDTO
    {
        public long UserId { get; set; }
        public long PlanId { get; set; }
        public bool AutoRenew { get; set; } = true;
    }

    public record class GetSubscriptionDTO : BaseDTO
    {
        public long UserId { get; set; }
        public long PlanId { get; set; }
        public SubscriptionStatus Status { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public DateTime? NextBillingDate { get; set; }
        public bool AutoRenew { get; set; }
        public DateTime? CancelledAt { get; set; }
        public string? CancellationReason { get; set; }
    }

    public record class CancelSubscriptionDTO
    {
        public Guid SubscriptionUid { get; set; }
        public string? CancellationReason { get; set; }
    }

    public record class GetPlanDTO : BaseDTO
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public decimal Price { get; set; }
        public string Currency { get; set; } = string.Empty;
        public string BillingPeriod { get; set; } = string.Empty;
        public bool IsActive { get; set; }
        public int? MaxBoards { get; set; }
        public int? MaxMembers { get; set; }
        public int? AIGenerationsPerMonth { get; set; }
    }

    public record class AddPaymentMethodDTO
    {
        public long UserId { get; set; }
        public string Type { get; set; } = string.Empty;
        public string Provider { get; set; } = string.Empty;
        public string Token { get; set; } = string.Empty;
        public string? LastFourDigits { get; set; }
        public DateTime? ExpiryDate { get; set; }
        public bool IsDefault { get; set; } = false;
    }

    public record class GetPaymentMethodDTO : BaseDTO
    {
        public long UserId { get; set; }
        public string Type { get; set; } = string.Empty;
        public string Provider { get; set; } = string.Empty;
        public string? LastFourDigits { get; set; }
        public DateTime? ExpiryDate { get; set; }
        public bool IsDefault { get; set; }
        public bool IsActive { get; set; }
    }
}

File: ClimbEdge.App/shared/src/types/PaymentDTO.ts
import { BaseDTO } from './BaseDTO';

export enum SubscriptionStatus {
  Active = 0,
  Cancelled = 1,
  Expired = 2,
  PastDue = 3,
}

export interface CreateSubscriptionDTO {
  userId: number;
  planId: number;
  autoRenew: boolean;
}

export interface CancelSubscriptionDTO {
  subscriptionUid: string;
  cancellationReason?: string;
}

export class GetSubscriptionDTO extends BaseDTO {
  userId!: number;
  planId!: number;
  status!: SubscriptionStatus;
  startDate!: string;
  endDate?: string | null;
  nextBillingDate?: string | null;
  autoRenew!: boolean;
  cancelledAt?: string | null;
  cancellationReason?: string;
}

export class GetPlanDTO extends BaseDTO {
  name!: string;
  description?: string;
  price!: number;
  currency!: string;
  billingPeriod!: string;
  isActive!: boolean;
  maxBoards?: number | null;
  maxMembers?: number | null;
  aiGenerationsPerMonth?: number | null;
}

export interface AddPaymentMethodDTO {
  userId: number;
  type: string;
  provider: string;
  token: string;
  lastFourDigits?: string;
  expiryDate?: string | null;
  isDefault: boolean;
}

export class GetPaymentMethodDTO extends BaseDTO {
  userId!: number;
  type!: string;
  provider!: string;
  lastFourDigits?: string;
  expiryDate?: string | null;
  isDefault!: boolean;
  isActive!: boolean;
}
